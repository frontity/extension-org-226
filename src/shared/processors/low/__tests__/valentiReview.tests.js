import Component from '../../../components/ValentiReview';
import main, {
  getTitle,
  getSummary,
  getCriteria,
  getUserRating,
} from '../valentiReview';
import data from './data/valentiReview.json';

describe('Extension Organization 226 › Shared › Processors › low › valentiReview', () => {
  test('extracts title ', () => {
    expect(getTitle(data)).toMatchSnapshot();
  });
  test('extracts summary data ', () => {
    expect(getSummary(data)).toMatchSnapshot();
  });
  test('extracts all criteria ', () => {
    expect(getCriteria(data)).toMatchSnapshot();
  });
  test('extracts user rating', () => {
    expect(getUserRating(data)).toMatchSnapshot();
  });
  test('processor works as expected', () => {
    const { test: t, process: p } = main;
    expect(t(data)).toBeTruthy();
    const processed = p(data);
    expect(processed.component).toBe(Component);
    expect(processed).toMatchSnapshot();
  });
});
