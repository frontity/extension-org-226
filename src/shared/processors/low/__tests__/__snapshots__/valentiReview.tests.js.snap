// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extension Organization 226 › Shared › Processors › low › valentiReview extracts all criteria  1`] = `
Array [
  Object {
    "label": "Precios",
    "stars": "12%",
  },
  Object {
    "label": "Métodos de pago",
    "stars": "12%",
  },
  Object {
    "label": "Diseño y usabilidad",
    "stars": "8%",
  },
  Object {
    "label": "Información al cliente",
    "stars": "8%",
  },
  Object {
    "label": "Servicios",
    "stars": "26%",
  },
]
`;

exports[`Extension Organization 226 › Shared › Processors › low › valentiReview extracts summary data  1`] = `
Object {
  "label": "Nota Final",
  "score": "8.7",
  "stars": "13%",
}
`;

exports[`Extension Organization 226 › Shared › Processors › low › valentiReview extracts title  1`] = `"Avianca Colombia: Opiniones y valoración"`;

exports[`Extension Organization 226 › Shared › Processors › low › valentiReview extracts user rating 1`] = `
Object {
  "label": "Puntuación de los lectores ",
  "stars": "100%",
  "votes": "0 Votos",
}
`;

exports[`Extension Organization 226 › Shared › Processors › low › valentiReview processor works as expected 1`] = `
Object {
  "component": [Function],
  "props": Object {
    "criteria": Array [
      Object {
        "label": "Precios",
        "stars": "12%",
      },
      Object {
        "label": "Métodos de pago",
        "stars": "12%",
      },
      Object {
        "label": "Diseño y usabilidad",
        "stars": "8%",
      },
      Object {
        "label": "Información al cliente",
        "stars": "8%",
      },
      Object {
        "label": "Servicios",
        "stars": "26%",
      },
    ],
    "summary": Object {
      "label": "Nota Final",
      "score": "8.7",
      "stars": "13%",
    },
    "title": "Avianca Colombia: Opiniones y valoración",
    "userRating": Object {
      "label": "Puntuación de los lectores ",
      "stars": "100%",
      "votes": "0 Votos",
    },
  },
}
`;
